src/pages/groupComparison/ClinicalData.tsx

```diff
@@ -338,17 +346,54 @@ export default class ClinicalData extends React.Component<
             const axisData = this.clinicalDataPromise.result!;
             // for string like values include NA values if requested
             // TODO: implement for numbers
+            // Compare clinical data(axisData) samples with sample list, assign NA to difference
+            // If exclude overlapped patients and samples, remove overlapped samples from sample list
+            // If include overlapped patients and samples, keep all sample in sample list
+            let filteredSamples: Sample[] = [];
+            if (this.props.store.overlapStrategy === OverlapStrategy.EXCLUDE) {
+                const overlappingPatients = getOverlappingPatients(
+                    this.props.store._activeGroupsNotOverlapRemoved.result!
+                );
+                const overlappingPatientsMap: {
+                    [studyId: string]: Set<string>;
+                } = {};
+                overlappingPatients.map(p => {
```

Comment:
user _.groupBy instead

Reply:
actually, it's not totally clear to me why you have to put them into groups by study.  they should have unique ID which should allow you to filter them without regard for study?